oshi-core/src/main/java/oshi/hardware/CentralProcessor.java

```diff
@@ -453,4 +458,84 @@ public int getProcessorGroup() {
             return processorGroup;
         }
     }
+
+
+    public static final class ProcessorIdentifier {
+        // Provided in constructor
+        private final String cpuVendor;
+        private final String cpuName;
+        private final String cpuFamily;
+        private final String cpuModel;
+        private final String cpuStepping;
+        private final String processorID;
+        private final String cpuIdentifier;
+        private final boolean cpu64bit;
+        private final long cpuVendorFreq;
+
+        public ProcessorIdentifier(String cpuVendor, String cpuName, String cpuFamily, String cpuModel,
+                String cpuStepping, String processorID, boolean cpu64bit) {
+            this.cpuVendor = cpuVendor;
+            this.cpuName = cpuName;
+            this.cpuFamily = cpuFamily;
+            this.cpuModel = cpuModel;
+            this.cpuStepping = cpuStepping;
+            this.processorID = processorID;
+            this.cpu64bit = cpu64bit;
+
+            // Build Identifier
+            StringBuilder sb = new StringBuilder();
+            if (cpuVendor.contentEquals("GenuineIntel")) {
+                sb.append(cpu64bit ? "Intel64" : "x86");
+            } else {
+                sb.append(cpuVendor);
+            }
+            sb.append(" Family ").append(cpuFamily);
+            sb.append(" Model ").append(cpuModel);
+            sb.append(" Stepping ").append(cpuStepping);
+            this.cpuIdentifier = sb.toString();
+
+            // Parse Freq from name string
+            Pattern pattern = Pattern.compile("@ (.*)$");
+            Matcher matcher = pattern.matcher(cpuName);
+            if (matcher.find()) {
+                String unit = matcher.group(1);
+                this.cpuVendorFreq = ParseUtil.parseHertz(unit);
+            } else {
+                this.cpuVendorFreq = -1L;
+            }
+        }
+
+        public String getVendor() {
```

Comment:
Good, but these need javadocs. You can copy the ones from above that you are deprecating.